src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,779 @@
+        if tags:
+            try:
+                for tag, value in tags.items():
+                    self.mlflow_client.set_registered_model_tag(
+                        name=name,
+                        key=tag,
+                        value=value,
+                    )
+            except MlflowException as e:
+                raise RuntimeError(
+                    f"Failed to update tags for the model {name} in MLflow model "
+                    f"registry: {str(e)}",
+                )
+        # Remove tags from the registered model.
+        if remove_tags:
+            try:
+                for tag in remove_tags:
+                    self.mlflow_client.delete_registered_model_tag(
+                        name=name,
+                        key=tag,
+                    )
+            except MlflowException as e:
+                raise RuntimeError(
+                    f"Failed to remove tags for the model {name} in MLflow model "
+                    f"registry: {str(e)}",
+                )
+        # Return the updated registered model.
+        return self.get_model(name)
+
+    def get_model(self, name: str) -> RegisteredModel:
+        """Get a model from the MLflow model registry.
+
+        Args:
+            name: The name of the model.
+
+        Returns:
+            The model.
+
+        Raises:
+            RuntimeError: If mlflow fails to get the model.
+        """
+        # Get the registered model.
+        try:
+            registered_model = self.mlflow_client.get_registered_model(
+                name=name,
+            )
+        except MlflowException as e:
+            raise RuntimeError(
+                f"Failed to get model with name {name} from the MLflow model "
+                f"registry: {str(e)}",
+            )
+        # Return the registered model.
+        return RegisteredModel(
+            name=registered_model.name,
+            description=registered_model.description,
+            tags=registered_model.tags,
+        )
+
+    def list_models(
+        self,
+        name: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> List[RegisteredModel]:
+        """List models in the MLflow model registry.
+
+        Args:
+            name: A name to filter the models by.
+            tags: The tags to filter the models by.
+
+        Returns:
+            A list of models (RegisteredModel)
+        """
+        # Set the filter string.
+        filter_string = ""
+        if name:
+            filter_string += f"name='{name}'"
+        if tags:
+            for tag, value in tags.items():
+                if filter_string:
+                    filter_string += " AND "
+                filter_string += f"tags.{tag}='{value}'"
+
+        # Get the registered models.
+        registered_models = self.mlflow_client.search_registered_models(
+            filter_string=filter_string,
+            max_results=100,
+        )
```

Comment:
I somehow went with making it open to any model that was registered within the used mlflow registry, because I thought these should also handle cases of registered models via UI